### PR TITLE
Get rid of char group strings

### PR DESCRIPTION
**Pull request by daschuer**
For accessing control objects we need QStrings and converting them on the fly
takes some unnecessary time.

I have stumbled over various time and now I finally find the mood to fix it. 

This is targeted to master, but branched from 2.3. 
So we can decide.

### CODE REVIEW

**Review by Holzhaus — COMMENTED** [review]
Thanks for taking care of this. IMHO this can go into 2.3.

**Review by Holzhaus — APPROVED** [review]
LGTM.

**Comment by uklotzde** [comment]
signalpathtest.h contains the last occurrence of `const char* group`

**Comment by daschuer** [comment]
Done.

File: src/test/signalpathtest.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ using ::testing::_;
 class TestEngineMaster : public EngineMaster {
   public:
     TestEngineMaster(UserSettingsPointer _config,
-            const char* group,
+            const QString group,
```
Comment (uklotzde) [inline src/test/signalpathtest.h:39]: either pass by value or by const-ref

Comment (daschuer) [reply src/test/signalpathtest.h]: Ups, fixed and force pushed.

**Review by Holzhaus — APPROVED** [review]
LGTM

**Comment by uklotzde** [comment]
@Holzhaus Suddenly the pre-commit hook starts complaining about the generated Rekordbox files. Any idea why?

**Comment by Holzhaus** [comment]
No, but we can exclude it via the pre-commit config.

**Comment by uklotzde** [comment]
@Holzhaus I'll amend the pre-commit config in another PR.

Still LGTM. Ready.

**Pull request merged**